src/Psalm/Internal/Type/Comparator/ScalarTypeComparator.php

```diff
@@ -550,4 +525,115 @@ public static function isContainedBy(
 
         return false;
     }
+
+    private static function isContainedByArrayKey(
+        Scalar $input_type,
+        TArrayKey $container_type,
+        Codebase $codebase
+    ): bool {
+        $container_type_class = get_class($container_type);
+        if ($container_type_class === TArrayKey::class) {
+            return $input_type instanceof TArrayKey
+                || $input_type instanceof TString
+                || $input_type instanceof TInt
+                || $input_type instanceof TNumeric;
+        }
+        if ($container_type instanceof TKeyOf) {
+            $key_types = TKeyOf::getArrayKeyType($container_type->type);
+            if ($key_types) {
+                foreach ($key_types->getAtomicTypes() as $key_type) {
+                    if (AtomicTypeComparator::isContainedBy($codebase, $input_type, $key_type)) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+        throw new Exception('Unexpected container type class: ' . $container_type_class);
```

Comment:
This needs to be less cryptic. If this is reported to us in the future, we need to know in which method this container was found to be able to fix